### PR TITLE
Add support for Nucleo L0

### PR DESCRIPTION
**Pull request by stephen99scott**
It doesn't seem like the system time functionality is working as expected

### CODE REVIEW

File: firmware/applications/SignBuddy/disco_l4_board.h [hunk 1]
```diff
@@ -69,7 +69,7 @@
 #define LOG_UART_GPIO_CLK_EN()        LL_AHB2_GRP1_EnableClock(LL_AHB2_GRP1_PERIPH_GPIOD)
 
 /* ADC */
-#define ADC                           ADC1
+#define ADC_PERIPH                    ADC1
```
Comment (adambujak) [inline firmware/applications/SignBuddy/disco_l4_board.h:72]: maybe name all of this to FLEX_ADC

**Comment by adambujak** [comment]
I don't think we should continue supporting the L4... it's a completely different SDK and we're just asking for trouble

File: firmware/applications/SignBuddy/disco_l4_board.h (hunk 1 above)
Comment (stephen99scott) [reply firmware/applications/SignBuddy/disco_l4_board.h:72]: Yup I'm good with that

**Comment by stephen99scott** [comment]
> I don't think we should continue supporting the L4... it's a completely different SDK and we're just asking for trouble

Yeah I agree, I just left it in for now because I wasn't sure if you guys were still gonna be using the L4 chip. I can remove the L4 stuff and restructure again if necessary

**Review by adambujak — APPROVED** [review]
I think this looks good. @harneetMac and/or @stephen99scott should review my changes before merge

**Comment by stephen99scott** [comment]
There's a bug with the adc readings, investigating now

**Comment by stephen99scott** [comment]
BLE uart not working

File: firmware/libraries/system_time/system_time.c [hunk 2]
```diff
@@ -40,22 +53,33 @@ inline uint32_t system_time_cmp_us(uint32_t old_time, uint32_t new_time)
...
   timer_init();
 }
+
+void SYSTEM_TIME_TIMER_IRQHandler(void)
+{
+  DISABLE_IRQ();
+  LL_TIM_ClearFlag_UPDATE(SYSTEM_TIME_TIMER);
+  s.overrun_cnt++;
+  ENABLE_IRQ();
+}
```
Comment (stephen99scott) [inline firmware/libraries/system_time/system_time.c:85]: I think there is potentially a conflict in the error handler functionality in main now.

The error handler disables irq, but also tries to call delay_ms which is dependant on irq

Comment (adambujak) [reply firmware/libraries/system_time/system_time.c:85]: good catch

File: firmware/applications/SignBuddy/main.c [hunk 3]
```diff
@@ -99,7 +96,7 @@ int main(void)
 
   LOG_INFO("App started\r\n");
 
-
+  error_handler();
```
Comment (adambujak) [inline firmware/applications/SignBuddy/main.c]: I forgot to remove this

**Pull request closed without merging**